### PR TITLE
Add interaction with meta data

### PR DESCRIPTION
**Pull request by breathx**

### CODE REVIEW

File: gstd/Cargo.toml [hunk 1]
```diff
@@ -7,6 +7,9 @@ license = "GPL-3.0"
 
 [dependencies]
 dlmalloc = { git = "https://github.com/nikvolf/dlmalloc-rs", features = ["global"] }
+scale-info = { version = "0.10.0", default-features = false, features = ["derive", "serde"] }
```
Comment (NikVolf) [inline gstd/Cargo.toml:10]: Nothing of this should be a dependency of gstd 

this should go to gstd-meta or similar

**Review by NikVolf — CHANGES_REQUESTED** [review]
Other than moving it all to separate crate lgtm

File: gstd-meta/codegen/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+extern crate alloc;
+extern crate proc_macro;
+
+use alloc::string::{String, ToString};
+use proc_macro::TokenStream;
+
+#[proc_macro_attribute]
+pub fn gear_data(_attr: TokenStream, item: TokenStream) -> TokenStream {
+    let derive_string = String::from("#[derive(Debug, Decode, Encode, TypeInfo)]");
```
Comment (NikVolf) [inline gstd-meta/codegen/src/lib.rs:29]: this is not needed (the whole `#[gear_data]` thing)

user is quite capable of putting `#[derive(Debug, Decode, Encode, TypeInfo)]` on his own

**Comment by NikVolf** [comment]
The js-test with meta should be extended to test dynamically generated type info-s

**Comment by breathx** [comment]
Moved to separate PR's with fixes

**Pull request closed without merging**